version: "3.9"

services:
  db:
    image: postgres:16
    environment:
      POSTGRES_DB: mydb
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: mypassword
    env_file:
      - ./.env
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    networks:
      - pay-first-network

  web:
    image: pay-first-web-app
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/app
    ports:
      - "8000:8000"
      - "5678:5678" # Debugpy port
    depends_on:
      - db
    env_file:
      - ./.env
    networks:
      - pay-first-network
    # Start debugpy for VS Code Python debugging
    entrypoint: ./entrypoint.sh
  tests:
    build: .
    command: python -m pytest -v
    volumes:
      - .:/app
    ports:
      - "5679:5678" # Debugpy port
    restart: always
    depends_on:
      - db
    env_file:
      - ./.env
    networks:
      - pay-first-network
    # Start debugpy for VS Code Python debugging
    entrypoint: ./entrypoint-test-cases.sh

volumes:
  postgres_data:
networks:
  pay-first-network:
    external: true
